src/components/LoginPage/index.js

```diff
@@ -42,18 +56,45 @@ class LoginPage extends React.Component {
       password
     };
 
-    this.setState({
-      loading: true
-    });
+    if (!email || !password) {
+      // if user tries to submit empty email/password
+      this.setState({
+        error: 'Please enter your email and password'
+      });
+    } else {
+      if (this.validateEmail(email)) {
+        this.setState({
+          loading: true
+        });
 
-    axios
-      .post(`${API_BASE_URL}/users/login`, userCredentials)
-      .then((res) => {
-        if (res.data.status === 'success') {
-          this.setState({
-            loading: false
-          });
+        axios
```

Comment:
Just Suggesting;
I think this (_action that initiates posting)_ should be taken to the `actions` file. so that its initiated by a reducer, please look into that thank you

Reply:
But in its current implementation, the code does the job

Reply:
Hey @Mubangizi I'm suggesting that this PR is merged then the redux issue is handled in a subsequent PR.
Because this one is meant to implement the validation.